samples/boards/tdk_robokit1/src/main.c

```diff
@@ -15,13 +15,16 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(TDK_ROBOKIT1, CONFIG_SENSOR_LOG_LEVEL);
 
+/* The fifo size (2048) bytes plus 3 for the header */
+#define ICM42688_FIFO_BUF_LEN 2051
+
 /* Power of 2 number of stream buffers */
 #define N_BUFS 4
 
 RTIO_EXECUTOR_SIMPLE_DEFINE(r_exec);
 
 /* Show no over/under flows with a 4 buffer queue */
-RTIO_DEFINE(r4, (struct rtio_executor *)&r_exec, N_BUFS, N_BUFS);
+RTIO_DEFINE_WITH_MEMPOOL(r4, (struct rtio_executor *)&r_exec, N_BUFS, N_BUFS, N_BUFS, ICM42688_FIFO_BUF_LEN, 4);
```

Comment:
The block size will need to be power of 2 I believe for mempool, so we might lose a sample or whatever in the watermark, that's ok. 1024 bytes gets us 63 sample packets plus room for the header (16 bytes). Should be fine to do just that

Reply:
The logic rounds the size up in the declare macro.